lib/nolibc/include/string.h

```diff
@@ -66,7 +66,7 @@ char *strtok(char *restrict s, const char *restrict sep);
 char *strtok_r(char *restrict s, const char *restrict sep, char **restrict p);
 char *strndup(const char *str, size_t len);
 char *strdup(const char *str);
-char *strncat(char *dest, const char *src, size_t n);
+char *strncat(char *dest, const char *src, size_t n)__nonnull;
```

Comment:
```suggestion
char *strncat(char *dest, const char *src, size_t n) __nonnull;
```

Reply:
Remove the `__nonnull` suffix here. It will be confusing why we're only using it for `strncat`. This requires a larger PR to add the `__nonnull` suffix to many more other functions.